src/engine/DivideStrategy.h

```diff
@@ -25,6 +25,8 @@ enum DivideStrategy
     Polarity,      // Pick the ReLU with the polarity closest to 0 among the first K nodes
     EarliestReLU,  // Pick a ReLU that appears in the earliest layer
     ReLUViolation, // Pick the ReLU that has been violated for the most times
+
+    Auto
 };
 
 #endif // __DivideStrategy_h__
```

Comment:
Looking at the code, there seems to be a mix here between two kinds of strategies: one for the DnC splitting, and one for the internal Marabou splitting. I recommend separating the options into two separate enums - otherwise it's very confusing, when some cases are not handled in the code. Polarity can appear in both enums if needed (one could conceivably do LargestInterval for DnC, and polarity for internal Marabou splitting, right?)

Reply:
Created a new enum class called SnCDivideStrategy; add methods in Engine pickSplittingPLConstraintSnC() which is a call-back for QueryDividers.